test/sample_test.go

```diff
@@ -0,0 +1,57 @@
+package test
+
+import (
+	"fmt"
+	"testing"
+
+	k "github.com/opendevstack/pipeline/internal/kubernetes"
+	"github.com/opendevstack/pipeline/internal/tekton"
+)
+
+func TestSatisfiedBy(t *testing.T) {
+
+	clients := k.NewClients()
+	taskName := "hello-world"
+	storageCapacity := "1Gi"
+	storageClassName := "" //"standard"
+
+	tests := map[string]struct {
+		sourceDirectory string
+		workspaceName   string
+		claimName       string
+		params          map[string]string
+		wantSuccess     string
+	}{
+		"task should return success": {
+			sourceDirectory: "./go-app",
+			workspaceName:   "source", // must exist in the Task definition

```

Comment:
I wonder if we actually need a mapping here. E.g. a task might define more than one workspace. maybe a map of workspaceName/sourceDirectory would fit our needs well?

Reply:
Fixed